### PR TITLE
Add org.tropy.Tropy

### PR DESCRIPTION
**Pull request by inukshuk**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. Home dir access included because the app needs to access existing Tropy projects.
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project.
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: org.tropy.Tropy.metainfo.xml
Comment (bbhtt) [inline org.tropy.Tropy.metainfo.xml:1]:
```
→ appstream-util validate test.appdata.xml
test.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
Validation of files failed
```

This needs to be fixed. Also this file should ideally live upstream

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (bbhtt) [inline flathub.json:2]: aarch64 is not available?

File: org.tropy.Tropy.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  - --talk-name=com.canonical.AppMenu.Registrar
+  - --talk-name=com.canonical.indicator.application
+  - --talk-name=com.canonical.Unity.LauncherEntry
+
+modules:
+  - name: tropy
+    buildsystem: simple
+    sources:
+      - type: archive
+        url: https://github.com/tropy/tropy/releases/download/v1.14.0/tropy-1.14.0-x64.tar.bz2
```
Comment (hfiguiere) [inline org.tropy.Tropy.yml:34]: does this include fixes for the webp vulnerability?

File: tropy.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+zypak-wrapper /app/tropy/tropy "$@"
```
Comment (hfiguiere) [inline tropy.sh:3]:
```suggestion
exec zypak-wrapper /app/tropy/tropy "$@"
```

File: org.tropy.Tropy.yml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      - type: file
+        path: org.tropy.Tropy.metainfo.xml
+      - type: file
+        path: tropy.sh
+    build-commands:
+      - mkdir -p /app/bin /app/share/{applications,metainfo}
+      - install -m644 -t /app/share/metainfo org.tropy.Tropy.metainfo.xml
+      - install -m755 tropy.sh /app/bin/tropy
+      - sed -i 's/Icon=tropy/Icon=org.tropy.Tropy/' tropy/tropy.desktop
+      - install -m644 tropy/tropy.desktop /app/share/applications/org.tropy.Tropy.desktop
```
Comment (hfiguiere) [inline org.tropy.Tropy.yml:48]:
```suggestion
      - install -Dm644 tropy/tropy.desktop /app/share/applications/org.tropy.Tropy.desktop
```

File: org.tropy.Tropy.yml (hunk 2 above)
Comment (bbhtt) [reply org.tropy.Tropy.yml:34]: Uses 24.2.0 https://github.com/tropy/tropy/blob/0afaddaddbbc60ea2982afc5331981f53d8e89c8/package.json#L68C19-L68C25, 24.8.3 is the fixed version is v24 branch afaik.

File: org.tropy.Tropy.metainfo.xml
Comment (inukshuk) [reply org.tropy.Tropy.metainfo.xml:1]: Oh, thanks! We'll add it upstream.

File: org.tropy.Tropy.yml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+  - --device=dri
+  - --filesystem=home
+  - --filesystem=xdg-desktop
+  - --filesystem=xdg-documents
+  - --filesystem=xdg-download
+  - --filesystem=xdg-pictures
+  - --filesystem=xdg-public-share
```
[the remark this answers is not in the document]
Comment (inukshuk) [reply org.tropy.Tropy.yml:22]: I'll run some more tests. We might be able to switch to read-only access as long as the database file itself is picked via the dialog.

File: org.tropy.Tropy.yml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        dest: tropy
+      - type: file
+        path: org.tropy.Tropy.metainfo.xml
+      - type: file
+        path: tropy.sh
+    build-commands:
+      - mkdir -p /app/bin /app/share/{applications,metainfo}
+      - install -m644 -t /app/share/metainfo org.tropy.Tropy.metainfo.xml
+      - install -m755 tropy.sh /app/bin/tropy
+      - sed -i 's/Icon=tropy/Icon=org.tropy.Tropy/' tropy/tropy.desktop
```
[the remark this answers is not in the document]
Comment (inukshuk) [reply org.tropy.Tropy.yml]: We updated the file upstream to use the id for all icons

File: flathub.json (hunk 1 above)
Comment (inukshuk) [reply flathub.json]: I'm trying to build everything from source now and this looks good so far: we build libvips and dependencies fine, but when I try to run npm install inside the sandbox it fails with ENOENT for `/run/build/tropy/package.json`. If I drop in to the build with `--build-shell` and try to run `npm install` myself it runs for a while then fails with EAI_AGAIN. Do you know why that is?

**Comment by hfiguiere** [comment]
bot, build  org.tropy.Tropy

**Comment by hfiguiere** [comment]
bot, build org.tropy.Tropy

File: org.tropy.Tropy.yml [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+      - type: archive
+        url: https://github.com/tropy/tropy/archive/refs/tags/v1.15.1.tar.gz
+        sha256: e4d1b7a36f5cc0c7cd1621b713d7aeb6a6988aa935d4e4d34a488789c85a9e62
+        dest: tropy
+      - type: file
+        path: tropy.sh
+    build-commands:
+      - install -Dm755 tropy.sh /app/bin/tropy
+      - cd tropy
+      - npm install --no-progress --no-audit --no-fund
```
Comment (hfiguiere) [inline org.tropy.Tropy.yml:177]: where does it get the packages?

Probably need to use https://github.com/flatpak/flatpak-builder-tools to generate the list.

Comment (inukshuk) [reply org.tropy.Tropy.yml:177]: Ah, I see thanks!

File: org.tropy.Tropy.yml [hunk 8]
```diff
@@ -13,14 +13,16 @@ finish-args:
...
   #- --socket=fallback-x11
   - --socket=x11
-  - --socket=pulseaudio
+  #- --socket=pulseaudio
   - --share=network
   - --device=dri
   - --filesystem=home
   - --filesystem=/media
   - --filesystem=/run/media
+  - --filesystem=/tmp
```
Comment (bbhtt) [inline org.tropy.Tropy.yml:22]: Why do you need access to real temp? Can setting the TMPDIR variable work?

Comment (inukshuk) [reply org.tropy.Tropy.yml:22]: Yes probably so. I had issues with the Chromium lockfiles and thought that this fixed it at some point.

**Comment by hfiguiere** [comment]
it's electron don't bother with wayland.

**Comment by bbhtt** [comment]
bot, build org.tropy.Tropy

**Comment by inukshuk** [comment]
Wayland seems to work fine with our client side decorations, except for missing shadows of the menus (I may have overlooked other issues of course). However, I think we'll stick with X11/Xwayland for the time being until this is safer.

Otherwise is there a chance we can move this to Flathub as is or would you prefer more changes?

**Pull request closed without merging**